pkgs/by-name/st/styluslabs-write3/package.nix

```
@@ -0,0 +1,112 @@
+  SDL2,
+  xorg,
+  dbus,
+  libcap,
+  libGL,
+  roboto,
+  imagemagick,
+  makeDesktopItem,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "Write3";
+  version = "2024-09-26";
+
+  src = fetchFromGitHub {
+    owner = "styluslabs";
+    repo = "Write";
+    rev = "7110e5db87c154d28efd9507be4ab768af7403e0";
+    hash = "sha256-/IYI/trC14IBYtYn9LkjNrMZoCn6jtQIgq0ZqnsLISs=";
+    fetchSubmodules = true;
+  };
+
+  postPatch = ''
+    # export SDL_LIBS=$(sdl2-config --static-libs)
+    export SDL_LIBS="$(sdl2-config --libs) -lX11 -lXcursor -lXi"
+    substituteInPlace syncscribble/Makefile \
+        --replace-fail 'LIBS = ../SDL/$(BUILDDIR)/libSDL2.a' "" \
+        --replace-fail 'LIBS += -pthread -ldl -lX11 -lXi' "LIBS += \$(SDL_LIBS)"
+    # cat syncscribble/Makefile
+  '';
+
+  preBuild = ''
+    # vendored statically linked SDL 2.24.1 as git submodule is not used
+    # https://github.com/pbsurf/SDL/commits/write-linux
+    # make -f ../scribbleres/SDL-Makefile.unix DEBUG=0 BUILDDIR=LinuxRel
+    # TODO: diff SDL_config_linux.h
+    rm -rf SDL/
+    cd syncscribble
+  '';
+
+  hardeningDisable = [ "format" ];
+  makeFlags = [
+    "DEBUG=0"
+    "GITREV=${finalAttrs.src.rev}"
+  ];
+  env.NIX_CFLAGS_COMPILE = "-I${lib.getDev SDL2}/include/SDL2";
+
+  nativeBuildInputs = [
+    imagemagick # magick
+    pkg-config
+    gitMinimal
+  ];
+
+  buildInputs = [
+    SDL2
+    # (SDL2.override { withStatic = true; })
+    xorg.libX11
+    xorg.libXi
+    xorg.libXrandr
+    xorg.libXcursor
+    xorg.libXfixes
+    xorg.libXScrnSaver
+    libGL
+    dbus.lib
+    libcap.lib
+  ];
+
+  desktopItem = makeDesktopItem {
+    name = "Write3";
+    desktopName = "Write3";
+    comment = "A word processor for handwriting";
+    mimeTypes = [ "image/svg+xml" ];
+    exec = "Write3";
+    icon = "Write3";
+    categories = [ "Office" ];
+  };
+
+  installPhase = ''
+    runHook preInstall
+    cd ..
+    install -m555 -D syncscribble/Release/Write $out/libexec/Write3
+    install -m444 -D scribbleres/Intro.svg $out/libexec/
+    ln -s ${roboto}/share/fonts/truetype/Roboto-Regular.ttf $out/libexec/Roboto-Regular.ttf
+
+    # install -m444 -D scribbleres/fonts/DroidSansFallback.ttf $out/libexec/
+    # install -m444 -D scribbleres/fonts/Roboto-Regular.ttf $out/libexec/
+    mkdir $out/bin
+    ln -s ../libexec/Write3 $out/bin/Write3
+
+    for i in 16 24 48 64 96 128 256 512; do
+      mkdir -p $out/share/icons/hicolor/''${i}x''${i}/apps
+      magick scribbleres/write_512.png -resize ''${i}x''${i} $out/share/icons/hicolor/''${i}x''${i}/apps/${finalAttrs.pname}.png
+    done
+
+    ln -s ${finalAttrs.desktopItem}/share/applications $out/share/
+  '';
+
+  meta = {
+    homepage = "https://styluslabs.com/";
+    description = "Cross-platform (Windows, Mac, Linux, iOS, Android) application for handwritten notes.";
+    license = lib.licenses.agpl3Only;
+    # maintainers = with lib.maintainers; [ lukts30 ];
+    platforms = lib.platforms.linux;
```

Platforms should state platform support of the package, whether the drv works for that platform or not is a matter of brokenness. 

```suggestion
    platforms = with lib.platforms; linux ++ darwin ++ windows;
    broken = !stdenv.isLinux;
```